src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1002,6 +1008,8 @@ public Consumer createConsumerFromDTO(ConsumerDTO consumer, ConsumerType type, P
             Date createdDate = consumerToCreate.getCreated();
             Date lastCheckIn = consumerToCreate.getLastCheckin();
 
+            this.updateEnvironments(consumer, consumerToCreate, false);
```

Comment:
Using this method for handling environments during consumer registration is fine, but you should remove the environment handling done in the populateEntity() method then, because now we're doing duplicate work.

Reply:
Done.